src/main/java/net/fabricmc/loader/api/config/serialization/toml/TomlReader.java

```diff
@@ -0,0 +1,680 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.api.config.serialization.toml;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+/**
+ * Class for reading TOML v0.4.0.
+ * <h1>DateTimes support</h1>
+ * <p>
+ * The datetime support is more extended than in the TOML specification. This reader supports three kind of datetimes:
+ * <ol>
+ * <li>Full RFC 3339. Examples: 1979-05-27T07:32:00Z, 1979-05-27T00:32:00-07:00, 1979-05-27T00:32:00.999999-07:00</li>
+ * <li>Without local offset. Examples: 1979-05-27T07:32:00, 1979-05-27T00:32:00.999999</li>
+ * <li>Without time (just the date). Example: 2015-03-20</li>
+ * </ol>
+ * Moreover, parsing datetimes gives different objects according to the informations provided. For example, 2015-03-20
+ * is parsed as a {@link LocalDate}, 2015-03-20T19:04:35 as a {@link LocalDateTime}, and 2015-03-20T19:04:35+01:00 as a
+ * {@link ZonedDateTime}.
+ * </p>
+ * <h1>Lenient bare keys</h1>
+ * <p>
+ * This library allows "lenient" bare keys by default, as opposite to the "strict" bare keys required by the TOML
+ * specification. Strict bare keys may only contain letters, numbers, underscores, and dashes (A-Za-z0-9_-). Lenient
+ * bare keys may contain any character except those below the space character ' ' in the unicode table, '.', '[', ']'
+ * and '='. The behaviour of TomlReader regarding bare keys is set in its constructor.
+ * </p>
+ *
+ * @author TheElectronWill
```

Comment:
You are going to have to remove the entire toml impl. Nightconfig is LGPL which would force fabric loader the same way.